Delcam.ProductInterface.PowerMILL/Classes/PMProject.cs

```diff
@@ -429,13 +429,16 @@ public void CreateBlock(string modelName, double blockExpansion)
         /// </summary>
         /// <param name="boundary">The boundary from which to create the block.</param>        
         public BoundingBox CreateBlockFromBoundary(PMBoundary boundary)
-        {            
-            _powerMILL.DoCommand("EDIT MODEL ALL DESELECT ALL");
-            _powerMILL.DoCommand(string.Format("ACTIVATE BOUNDARY \"{0}\"", boundary.Name));
-            _powerMILL.DoCommand("EDIT BLOCKTYPE BOUNDARY");
-            _powerMILL.DoCommand("EDIT BLOCK RESETLIMIT 0");
-            _powerMILL.DoCommand("EDIT BLOCK RESET");
-            _powerMILL.DoCommand("BLOCK ACCEPT");
+        {
+            if (boundary != null)
```

Comment:
This feels weird...  If the parameter is null then you should throw a [ArgumentNullException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=netframework-4.8).

If the real issue is that the PMBoundary doesn't exist in PowerMill then use the Exists function to check for it.

Reply:
We have to check both. Because if the user call the function in this way

`powermill.ActiveProject.CreateBlockFromBoundary(powermill.ActiveProject.Boundaries.GetByName("Test"))`

the boundary is null if the boundary doesn't exists